### PR TITLE
Add particle systems

### PR DESCRIPTION
**Pull request by nadult**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,11 +11,13 @@ GCC = g++
 endif
 LD = $(GCC)
 
-ifndef RELEASE
-CFLAGS += -Werror -Wimplicit-fallthrough
+ifdef LLD
```
Comment (nadult) [inline Makefile]: Something is wrong when compiling with latest clang (6.0) and linking with lld. I get following errors:
> defined in /usr/lib/x86_64-linux-gnu/libopenal.so
> referenced by audio_device.cpp:17
>               obj-opt/audio_device.o:(_ZN11AudioDevice10initializeB5cxx11Ev) /usr/lib/llvm-6.0/bin/ld.lld: error: cannot preempt symbol: alcCreateContext
> defined in /usr/lib/x86_64-linux-gnu/libopenal.so
> referenced by audio_device.cpp:18
>               obj-opt/audio_device.o:(_ZN11AudioDevice10initializeB5cxx11Ev)
...

Comment (miki151) [reply Makefile]: I tried out clang 6.0 and lld works fine for me, so I think it's something specific to your system, maybe the version of openal. In any case I think it's more convenient to use lld by default and add a flag like NO_LLD.

File: Makefile [hunk 2]
```diff
@@ -11,11 +11,13 @@ GCC = g++
...
 LDFLAGS += -fuse-ld=lld
 endif
 
+ifndef RELEASE
+CFLAGS += -Werror -Wimplicit-fallthrough -Wno-unused-private-field -Wno-unused-function
```
Comment (miki151) [inline Makefile]: Can you remove -Wno-unused-private-field? I removed the fields that were causing the error.

File: Makefile (hunk 1 above)
Comment (nadult) [reply Makefile]: OK, I'll see what's going on here and if it still won't work, I'll change it to NO_LLD.

File: Makefile (hunk 2 above)
Comment (nadult) [reply Makefile]: Sure.

File: Makefile [hunk 3]
```diff
@@ -11,12 +11,13 @@ GCC = g++
 endif
 LD = $(GCC)
 
+DEBUG_LD=lld
```
Comment (nadult) [inline Makefile]: Now it's possible to change default ld by passing DEBUG_LD=... (ifndef / endif is not really required).

**Comment by nadult** [comment]
All of the raised issues were addressed. Too bad that github cannot show differences between the rebases (there were a couple of those).

**Comment by nadult** [comment]
Ahh, few more variables required renaming. So, additional rebase incoming...

**Comment by nadult** [comment]
OK done :) Next time I'll try to do the rebase only after getting :+1: .

**Pull request merged**